Include/cpython/pystats.h

```diff
@@ -133,6 +133,9 @@ typedef struct _rare_event_stats {
     uint64_t builtin_dict;
     /* Modifying a function, e.g. func.__defaults__ = ..., etc. */
     uint64_t func_modification;
+    /* Modifying a dict that is being watched */
+    uint64_t watched_dict_modification;
+    uint64_t watched_globals_modification;
```

Comment:
How do these new stats relate to the existing `builtin_dict` one?

Reply:
They only apply to any watched dict, and watched module globals respectively.
Whereas the builtin dict count tracks any modification to the builtins, watched or not.

In theory `watched_globals_modification + builtin_dict == watched_dict_modification`
Although, as soon as we start watching any other dicts that will no longer be true.

Reply:
That makes sense, thanks.